network/middleware.go

```diff
@@ -73,7 +73,7 @@ type Overlay interface {
 	// GetIdentity returns the Identity associated with the given peer ID, if it exists
 	Identity(peer.ID) (*flow.Identity, bool)
 
-	Receive(nodeID flow.Identifier, msg *message.Message) error
+	Receive(nodeID flow.Identifier, msg *message.Message, decodedMsgPayload interface{}) error
```

Comment:
Do we actually need to keep the second parameter?

Reply:
Not necessarily, but we could possibly use it in the future. I think it's worth keeping, what do you think?